Intersect.Client.Core/Interface/Game/Spells/SpellItem.cs

```diff
@@ -250,36 +292,43 @@ public void Update()
         }
         else
         {
-            if (mDragIcon.Update())
+            if (_dragIcon?.Update() == true)
             {
                 //Drug the item and now we stopped
                 IsDragging = false;
+                _iconImage.IsVisibleInParent = true;
+
                 var dragRect = new FloatRect(
-                    mDragIcon.X - (Container.Padding.Left + Container.Padding.Right) / 2,
-                    mDragIcon.Y - (Container.Padding.Top + Container.Padding.Bottom) / 2,
-                    (Container.Padding.Left + Container.Padding.Right) / 2 + Pnl.Width,
-                    (Container.Padding.Top + Container.Padding.Bottom) / 2 + Pnl.Height
+                    _dragIcon.X - (Padding.Left + Padding.Right) / 2f,
+                    _dragIcon.Y - (Padding.Top + Padding.Bottom) / 2f,
+                    (Padding.Left + Padding.Right) / 2f + _iconImage.Width,
+                    (Padding.Top + Padding.Bottom) / 2f + _iconImage.Height
                 );
 
                 float bestIntersect = 0;
                 var bestIntersectIndex = -1;
 
                 //So we picked up an item and then dropped it. Lets see where we dropped it to.
                 //Check spell first.
-                if (mSpellWindow.RenderBounds().IntersectsWith(dragRect))
+                if (_spellWindow.RenderBounds().IntersectsWith(dragRect))
                 {
                     for (var i = 0; i < Options.Instance.Player.MaxSpells; i++)
                     {
-                        if (i < mSpellWindow.Items.Count &&
-                            mSpellWindow.Items[i].RenderBounds().IntersectsWith(dragRect))
+                        if (_spellWindow.Items[i] is not SpellItem spellSlot)
                         {
-                            if (FloatRect.Intersect(mSpellWindow.Items[i].RenderBounds(), dragRect).Width *
-                                FloatRect.Intersect(mSpellWindow.Items[i].RenderBounds(), dragRect).Height >
+                            continue;
+                        }
+
+                        if (i < _spellWindow.Items.Count &&
```

Comment:
Before the for loop even starts, the end should be calculated with `Math.Min(<count>, <max spells>)`, this part of the if condition is completely avoidable